hypertrace-trace-enricher/hypertrace-trace-enricher-api/src/main/java/org/hypertrace/traceenricher/trace/util/GraphBuilderUtil.java

```diff
@@ -56,4 +58,12 @@ static boolean isTraceEntitiesChanged(StructuredTrace cachedTrace, StructuredTra
     }
     return false;
   }
+
+  /** Check whether these spans belongs to different services. */
```

Comment:
I think this method should be in ApiBoundaryTypeAttributeEnricher itself, this class is housing utilities related rebuilding of ApiTraceGraph and StructuredTraceGraph

Reply:
This is used in ApiTraceGraph as well.

Reply:
I see its used in ApiTraceGraph as well, then the right place for it is utility for span/enricher span

Reply:
I didn't get it, which class is ideal for this ? In GraphQLBuilder util most of the methods are related to what i have added now, although name(GraphQLBuilderUtil) is not looking right

Reply:
> I didn't get it, which class is ideal for this 

`EnrichedSpanUtils` (most of the method for checking some property on span is present here)

> In GraphBuilder util most of the methods are related to what

most of the method here is to check the existing event/trace has changed or not and is meant to support the logic for Graph caching logic,  whereas the newly added method is unrelated to graph rebuilding logic 

yeah this class needs renaming

Reply:
Yes, in fact i realised while doing changes in Traceable, moved to EnrichedSpanUtils.